dts: bindings: Add Xen Platform related dts bindings

povergoing opened the pull request:
Add the Xen Platform dts bindings.

fixes: https://github.com/zephyrproject-rtos/zephyr/pull/46494#issuecomment-1154687751

Temply set `CONFIG_LOG_MODE_MINIMAL=n` and `CONFIG_USERSPACE=n` to fix the build error.
cc: @firscity @luca-fancellu

Signed-off-by: Jaxson Han <jaxson.han@arm.com>

povergoing committed:
dts: bindings: Add Xen Platform related dts bindings

Add the Xen Platform related dts bindings.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>

stephanosio commented on boards/arm64/xenvm/xenvm_gicv3.yaml:
```diff
@@ -0,0 +1,8 @@
+toolchain:
+  - zephyr
+  - cross-compile
+ram: 16
```
```suggestion
ram: 16384
```

povergoing replied:
fixed. Always forget this is KB

carlocaione commented on boards/arm64/xenvm/xenvm.dts:
```diff
@@ -78,7 +78,8 @@
-		compatible = "xen,uart_hvc";
+		compatible = "xen,hvc-uart";
 		status = "okay";
+		label = "UART_HVC";
```
Is this used? We are trying to move away from labels.

povergoing replied:
There is a dependency loop I think. https://github.com/zephyrproject-rtos/zephyr/blob/1578dc699be8d334ec36123996922fef375bde81/dts/bindings/serial/uart-controller.yaml#L17 needs the label and the uart bindings includes `uart-controller.yaml`, so ...

But the @galak's cleanup work https://github.com/zephyrproject-rtos/zephyr/pull/46494#issuecomment-1154687751 depends on this one.

carlocaione replied:
Ok, if there is a dependency leave it and @galak can remove it later

povergoing replied:
Right

galak replied:
We need it for now.  Will look at cleaning the UART use of `label` up in the future, for now just add it.

galak commented on dts/bindings/serial/xen,hvc-uart.yaml:
```diff
@@ -0,0 +1,9 @@
+
+properties:
+    reg:
+      required: false
```
If this is `false` there is no need to have the `properties` section at all, since the default for `reg` property will be `required: false`

povergoing replied:
sure, done it

galak reviewed (changes requested):
minor comment on the uart yaml binding, otherwise LGTM.

povergoing committed:
boards: arm64: xenvm: Fix build issues

Temporarily set CONFIG_LOG_MODE_MINIMAL=n and CONFIG_USERSPACE=n to fix
the build error.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
